src/attitude-utils.cpp

```diff
@@ -469,6 +510,19 @@ decimal Distance(const Vec2 &v1, const Vec2 &v2) {
     return sqrt(pow(v1.x-v2.x, 2) + pow(v1.y-v2.y, 2));
 }
 
+/**
+ * Determines the  between two vectors
+ * 
+ * @param v1 The first vector
+ * @param v2 The second vector
+ * 
+ * @return The distance between v1 and v2
+ * 
+*/
+decimal Distance(const Vec3 &v1, const Vec3 &v2) {
+    return sqrt(pow(v1.x-v2.x, 2) + pow(v1.y-v2.y, 2) + pow(v1.z-v2.z, 2));
```

Comment:
Another way we can do this, is to compute the magnitude of the difference between the two vectors.

So, think about which one is faster & more memory efficient.